--- a/src/right-panel.js
+++ b/src/right-panel.js
@@ -94,6 +94,7 @@ export function init() {
 		if (snapshot())
 			qs('span', snap).innerText = "Update Analysis";
 	};
+	snap.remove();
 
 	const suid = maybe(SNAPSHOT, 'user_id');
 
@@ -114,6 +115,7 @@ export function init() {
 		if (snapshot(share_url))
 			qs('span', snap).innerText = "Update Analysis";
 	};
+	share.remove();
 
 	const download = qs('#tiff-download');
 	download.onclick = async _ => {
--- a/src/s.js
+++ b/src/s.js
@@ -29,11 +29,7 @@ function preload_boundaries(id) {
 };
 
 async function geography(c) {
-	const coll = await API.get("geographies", {
-		"datasets_count": "gt.0",
-		"parent_id":      `eq.${c.id}`,
-		"deployment":     `ov.{${ENV}}`,
-	});
+	const coll = [c, c.subgeographies].filter(x => x).flat();
 
 	if (c.datasets_count > 2) coll.unshift(c); // 2 datasets: outline and admin-tiers
 
@@ -231,15 +227,11 @@ export async function init() {
 
 			const _name = co.name.replace(new RegExp("\\ ?\\(?(" + ENV.join('|') + ")\\)?", "i"), '');
 
-			fetch(`${EAE['settings'].world}/countries?select=flag&or=(names->>official.eq."${_name}",name.eq."${_name}")`)
-				.then(r => r.json())
+			fetch(EAE['settings'].database + `/flags?name=eq.${co.name}`)
+				.then(r => r.text())
 				.then(r => {
-					const data = r[0];
-
-					if (!data) return;
-
 					d.append(ce('img', null, {
-						"src":   URL.createObjectURL((new Blob([hextostring(data['flag'])], {"type": "image/svg+xml"}))),
+						"src":   URL.createObjectURL((new Blob([r], {"type": "image/svg+xml"}))),
 						"class": "flag",
 					}));
 				});
